backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotResource.java

```diff
@@ -26,15 +26,15 @@ protected BackendDiskSnapshotResource(String id, BackendDiskSnapshotsResource pa
 
     @Override
     public DiskSnapshot get() {
-        // TODO: GetAllDiskSnapshots is an incorrect query to use here, it fetches all the disk snapshots, so the one
-        // returned here is not necessarily the one requested
-        DiskSnapshot diskSnapshot = performGet(QueryType.GetAllDiskSnapshots,
-                new DiskSnapshotsQueryParameters(diskId, true, false), Disk.class);
+        DiskSnapshot diskSnapshot = performGet(QueryType.GetDiskSnapshotByImageId,
+                new IdQueryParameters(guid), Disk.class);
         diskSnapshot.setDisk(new Disk());
         diskSnapshot.getDisk().setId(diskId.toString());
+        diskSnapshot.getDisk().setHref(backendDiskSnapshotsResource.buildParentHref(diskId.toString(), false));
```

Comment:
Not sure it's that critical, but why was is changed from buildHref to buildParentHref here, and vise versa on line 36?

Reply:
Here, the <disk> element didn't have an href attribute at all, although in the disks/xxx/disksnapshots request it has. Using buildParentHref because it's basically link to the parent disks/xxx.
As for the fix in line 36, it's not the same parent - it's the image parent, so buildParentHref is irrelevant here.

Reply:
Ack